source/state_representation/README.md

```diff
@@ -821,203 +827,618 @@ The inverse of a state will set the wrench to zero.
+frame and reference frame.
+
+```c++
+state_representation::Jacobian jacobian("my_robot", 3, "end_effector", "base_frame");
+jacobian.set_data(...); // set the Jacobian data matrix for the current robot configuration
+
+// get the Cartesian wrench of the same frame and reference frame
+auto end_effector_wrench = state_representation::CartesianTwist::Random("end_effector", "base_frame");
+
+// compute the torques of each joint from the Cartesian twist
+state_representation::JointTorques joint_torques = jacobian.transpose(end_effector_wrench);
+
+// the matrix product should only be used on raw data vectors
+Eigen::VectorXd raw_joint_torques = jacobian.transpose() * end_effector_wrench.get_wrench();
 ```
 
 ### Changing the Jacobian reference frame
 
-As stated earlier, the `Jacobian` is expressed in a reference frame and can, therefore, use the operations with
-`CartesianPose` to be modified. It relies on the usage of the overloaded `operator*` or the `set_reference_frame`
-function for inplace modifications. It is equivalent to multiply each columns of the `Jacobian` by the `CartesianPose`
-on both the linear and angular part of the matrix. For this operation to be valid, the `CartesianPose` name has to
-match the current `Jacobian` reference frame.
-
-```c++
-state_representation::Jacobian jac = state_representation::Jacobian::Random("myrobot", 3, "eef", "base_frame");
-state_representation::CartesianPose pose = state_representation::CartesianPose::Random("base_frame", "world");
-// the result is the Jacobian expressed in world
-state_representation::Jacobian jac_in_world = pose * jac;
-// alternatively, one case use the set_reference_frame function for inplace modifications
-jac.set_reference_frame(pose);
-// in case of non matching operation throw an IncompatibleStatesExceptions
-jac.set_reference_frame(state_representation::CartesianPose::Random("link0", "world"));
-```
\ No newline at end of file
+Whenever the `Jacobian` is used to convert between joint and Cartesian space, the Cartesian state variables must be
+expressed in the same reference frame as the `Jacobian`. If the `Jacobian` is expressed in a robot base frame while
+a `CartesianTwist` is measured in a world reference frame, then that twist cannot be directly used to find the
+corresponding joint velocities.
+
+```c++
+auto jacobian = state_representation::Jacobian::Random("my_robot", 3, "end_effector", "base_frame");
+auto twist_in_world = state_representation::CartesianTwist::Random("end_effector", "world");
+
+jacobian.pseudoinverse(twist_in_world); // error! IncompatibleStatesException
+```
+
+If the transformation between the robot base frame and world is known and expressed as a `CartesianPose`, it could be
+used to transform the `CartesianTwist` relative to the robot base frame (see the section describing
+[Cartesian transforms](#cartesian-transforms--changing-the-reference-frame) for more information).
```

Comment:
This double hyphen is correct here I guess? Does the`:` translate to a hyphen in a reference?

Reply:
It's how CLion markdown parser interpreted the anchor, but on the GitHub page it doesn't link through correctly. I think punctuation is generally ignored and spaces are replaced with hyphens.

- [cartesian-transforms--changing-the-reference-frame](#cartesian-transforms--changing-the-reference-frame)
- [cartesian-transforms-changing-the-reference-frame](#cartesian-transforms-changing-the-reference-frame)

### Cartesian transforms: changing the reference frame